internal/host/host.go

```diff
@@ -349,6 +350,22 @@ func (m *Manager) updateInventory(ctx context.Context, cluster *common.Cluster,
 		log.WithError(err).Errorf("not updating inventory - failed to check disks eligibility for host %s", h.ID)
 		return common.NewApiError(http.StatusInternalServerError, err)
 	}
+
+	// Remove virtual interfaces if the feature is not enabled
+	if !m.Config.EnableVirtualInterfaces {
+		var physicalInterfaces []*models.Interface = make([]*models.Interface, 0)
+		for _, intf := range inventory.Interfaces {
+			if intf.Type == "" {
+				// Indicates an older agent, so all interfaces are non-virtual
+				break
```

Comment:
If we break the loop here, when the (presumably) first interface doesn't have a type, won't we set `inventory.Interfaces` to the empty `physicalInterfaces` slice?

I think it would be better if you just treat the empty case the same as the other physical ones below and add it.

Also can you add a test case for an empty type?

Reply:
That's right, removed this break and added tests for the empty type. Thank you!